src/terminal/init.lua

```diff
@@ -331,6 +355,9 @@ end
 -- @treturn string ansi sequence to write to the terminal
 -- @within cursor_position
 function M.cursor_sets(row, column)
+  -- Resolve negative indices
+  row = resolve_index(row, M.get_height())
+  column = resolve_index(column, M.get_width())
```

Comment:
so this is what I meant with the inefficiencies.

2 functions calls, that will both call `sys.get_terminal_size`, each only using a single property. Instead call that function directly here, and we will save 50% of cpu cycles probably.